modules/local/misc.nf

```diff
@@ -20,11 +20,16 @@ process CAT_FASTQ {
 
     script:
     """
-    fastqFile=\$(ls $reads | head -1)
-    if grep -q "fastq.gz" <<< "\$fastqFile"; then
+    fastqFile=\$(ls $reads | grep -E ".fastq|fq" | head -1)
+    echo "\$fastqFile"
+    if grep -qE ".fastq.gz" <<< "\$fastqFile"; then
```

Comment:
Use *regular expression* please - that's what grep is made for. 

> grep is a command-line utility for searching plain-text data sets for lines that match a regular expression.
- https://www.gnu.org/software/grep/

See https://github.com/peterk87/nf-iav-illumina/pull/8#discussion_r872771682

This regex would reduce duplication of code:

```bash
grep -qE "\.(fastq|fq)\.gz$"
```
Note that periods are escaped and both "fastq" and "fq" are matched and "gz" must come at the end of the match with "$".

This process still needs to be wrapped in Groovy/Nextflow to check if `$reads` is a directory and if so then get all fastq files in that directory for concatenation. If instead, `$reads` is a file or collection of files (group by sample name/`meta.id`: `ch_reads.groupBy(by: 1)`). We commonly want to concatenate reads for the same sample sequenced multiple times to increase sequencing coverage.

Reply:
The cat command would need to be `cat $reads/*.{fq,fastq}.gz ...`. Curly braces and commas rather than brackets and pipes.